### PR TITLE
MM-54022: add warning when deleting a remote post

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
Adds a warning to the existing delete_post_dialog when post is remote/from a shared channel.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54022

#### Screenshots
Remote post
CleanShot 2024-09-23 at 10 16 40

Remote post with replies
CleanShot 2024-09-23 at 10 17 02

Remote comment
CleanShot 2024-09-23 at 10 51 41

#### Release Note

```release-note
Warn when deleting a post or comment from a remote/shared channel.
```

### CODE REVIEW

File: webapp/channels/src/components/delete_post_modal/delete_post_modal.tsx [hunk 1]
```diff
@@ -194,3 +202,30 @@ export default class DeletePostModal extends React.PureComponent<Props, State> {
...
     }
 }
+
+const SharedChannelPostDeleteWarning = ({post}: {post: Post}) => {
+    const {formatMessage} = useIntl();
+
+    const text = post.root_id ? (
+        formatMessage({
+            id: 'delete_post.shared_channel_warning.message_comment',
+            defaultMessage: 'This comment originated from a shared channel in another workspace, deleting it here will not remove it from the channel in the other workspace.',
```
Comment (cwarnermm) [inline webapp/channels/src/components/delete_post_modal/delete_post_modal.tsx:212]:
```suggestion
            defaultMessage: 'This comment originated from a shared channel in another workspace. Deleting it here won't remove it from the channel in the other workspace.',
```

File: webapp/channels/src/components/delete_post_modal/delete_post_modal.tsx [hunk 2]
```diff
@@ -194,3 +202,30 @@ export default class DeletePostModal extends React.PureComponent<Props, State> {
...
+        })
+    ) : (
+        formatMessage({
+            id: 'delete_post.shared_channel_warning.message_post',
+            defaultMessage: 'This post originated from a shared channel in another workspace, deleting it here will not remove it from the channel in the other workspace.',
```
Comment (cwarnermm) [inline webapp/channels/src/components/delete_post_modal/delete_post_modal.tsx:217]:
```suggestion
            defaultMessage: 'This message originated from a shared channel in another workspace. Deleting it here won't remove it from the channel in the other workspace.',
```

File: webapp/channels/src/i18n/en.json [hunk 3]
```diff
@@ -3451,6 +3451,9 @@
   "delete_post.del": "Delete",
   "delete_post.post": "Post",
   "delete_post.question": "Are you sure you want to delete this {term}?",
+  "delete_post.shared_channel_warning.message_comment": "This comment originated from a shared channel in another workspace, deleting it here will not remove it from the channel in the other workspace.",
```
Comment (cwarnermm) [inline webapp/channels/src/i18n/en.json:3454]:
```suggestion
  "delete_post.shared_channel_warning.message_comment": "This comment originated from a shared channel in another workspace. Deleting it here won't remove it from the channel in the other workspace.",
```

File: webapp/channels/src/i18n/en.json [hunk 4]
```diff
@@ -3451,6 +3451,9 @@
...
+  "delete_post.shared_channel_warning.message_post": "This post originated from a shared channel in another workspace, deleting it here will not remove it from the channel in the other workspace.",
```
Comment (cwarnermm) [inline webapp/channels/src/i18n/en.json:3455]:
```suggestion
  "delete_post.shared_channel_warning.message_post": "This message originated from a shared channel in another workspace. Deleting it here won't remove it from the channel in the other workspace.",
```

**Review by cwarnermm — APPROVED** [review]
Provided editorial feedback inline.

File: webapp/channels/src/components/delete_post_modal/__snapshots__/delete_post_modal.test.tsx.snap [hunk 5]
```diff
@@ -175,19 +177,16 @@ exports[`components/delete_post_modal should match snapshot for delete_post_moda
...
-    >
-      <MemoizedFormattedMessage
-        defaultMessage="This post has {count, number} {count, plural, one {comment} other {comments}} on it."
-        id="delete_post.warning"
-        values={
-          Object {
-            "count": 1,
-          }
+    <MemoizedFormattedMessage
+      defaultMessage="This post has {count, number} {count, plural, one {comment} other {comments}} on it."
```
Comment (cwarnermm) [inline webapp/channels/src/components/delete_post_modal/__snapshots__/delete_post_modal.test.tsx.snap:181]:
```suggestion
      defaultMessage="This message has {count, number} {count, plural, one {comment} other {comments}} on it."
```

File: webapp/channels/src/components/delete_post_modal/__snapshots__/delete_post_modal.test.tsx.snap [hunk 6]
```diff
@@ -63,6 +63,7 @@ exports[`components/delete_post_modal should match snapshot for delete_post_moda
     <MemoizedFormattedMessage
       defaultMessage="Are you sure you want to delete this {term}?"
```
Comment (cwarnermm) [inline webapp/channels/src/components/delete_post_modal/__snapshots__/delete_post_modal.test.tsx.snap:64]: Ideal: replace `{term]` as `post` with `message`

**Review by JulienTant — APPROVED** [review]

**Review by sbishel — APPROVED** [review]

**Comment by lindy65** [comment]
Hey @calebroseland - realised Connected Workspaces is not visible in the Sys Cons. Are you able to change the config on the spinwick or should I rather QA-defer and test post merge?

**Comment by calebroseland** [comment]
@lindy65 you may want to QA-defer this one. I don't know that I can get a spinwick up for this. I could otherwise update one of the other shared channels testing instances to this PR or spin up a new one if you want

**Comment by lindy65** [comment]
If you can update one of the existing test instances that would be great,
thanks 😊

**Comment by calebroseland** [comment]
Alright, https://cr-dev.test.mattermost.cloud is running this, now @lindy65. You may have to wait a few minutes, the server is currently being throttled.

**Comment by lindy65** [comment]
Thanks @calebroseland 🤗

**Review by lindy65 — APPROVED** [review]
Thanks @calebroseland - all scenarios look good to me

When you have time, please also update `https://shared-channels.test.mattermost.cloud/` with this code?  Thanks 🤗

**Pull request merged**